src/annotator/test/frame-observer-test.js

```diff
@@ -4,140 +4,191 @@ import {
   onDocumentReady,
 } from '../frame-observer';
 
-describe('FrameObserver', () => {
-  let container;
-  let frameObserver;
-  let onFrameAdded;
-  let onFrameRemoved;
-
-  function waitForFrameObserver() {
-    return new Promise(resolve => setTimeout(resolve, DEBOUNCE_WAIT + 10));
-  }
-
-  function createAnnotatableIFrame(
-    attribute = 'enable-annotation',
-    value = ''
-  ) {
-    const frame = document.createElement('iframe');
-    frame.setAttribute(attribute, value);
-    container.appendChild(frame);
-    return frame;
-  }
-
-  function waitForIFrameUnload(frame) {
-    return new Promise(resolve =>
-      frame.contentWindow.addEventListener('unload', resolve)
-    );
-  }
-
-  beforeEach(() => {
-    container = document.createElement('div');
-    document.body.appendChild(container);
-    onFrameAdded = sinon.stub();
-    onFrameRemoved = sinon.stub();
-    frameObserver = new FrameObserver(container, onFrameAdded, onFrameRemoved);
-  });
+describe('annotator/frame-observer.js', () => {
+  describe('FrameObserver', () => {
+    let container;
+    let frameObserver;
+    let onFrameAdded;
+    let onFrameRemoved;
+    const sandbox = sinon.createSandbox();
+
+    function waitForFrameObserver() {
+      return new Promise(resolve => setTimeout(resolve, DEBOUNCE_WAIT + 10));
+    }
+
+    function createAnnotatableIFrame(
+      attribute = 'enable-annotation',
+      value = ''
+    ) {
+      const iframe = document.createElement('iframe');
+      iframe.setAttribute(attribute, value);
+      container.appendChild(iframe);
+      return iframe;
+    }
+
+    function waitForIFrameUnload(iframe) {
+      return new Promise(resolve =>
+        iframe.contentWindow.addEventListener('unload', resolve)
+      );
+    }
+
+    beforeEach(() => {
+      container = document.createElement('div');
+      document.body.appendChild(container);
+      onFrameAdded = sandbox.stub();
+      onFrameRemoved = sandbox.stub();
+      frameObserver = new FrameObserver(
+        container,
+        onFrameAdded,
+        onFrameRemoved
+      );
+      sandbox.stub(console, 'warn');
+    });
 
-  afterEach(() => {
-    container.remove();
-    frameObserver.disconnect();
-  });
+    afterEach(() => {
+      container.remove();
+      frameObserver.disconnect();
+      sandbox.restore();
+    });
 
-  it('triggers onFrameAdded when an annotatable iframe is added', async () => {
-    let frame = createAnnotatableIFrame();
-    await waitForFrameObserver();
+    it('triggers onFrameAdded when an annotatable iframe is added', async () => {
+      let iframe = createAnnotatableIFrame();
```

Comment:
I'm not sure about the `frame` => `iframe` change here. We tend to use the term `frame` in most places (eg. in the sidebar) following the example of `window.frames` and other browser APIs.

Reply:
I have taken a middle ground here: `frame-observer.js` uses `frame` for the code (not for the comments) but `frame-observer-test.js` uses `iframe`. That way I leave the code as generic as possible, but the test that is the application on a concrete scenario, more specific.

Are you happy with this? Otherwise, I am happy to revert all to `frame`. Please, let me know.